crates/sui-graphql-rpc/src/extensions/timeout.rs

```diff
@@ -42,6 +44,13 @@ impl Extension for TimeoutExt {
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
         let document = next.run(ctx, query, variables).await?;
+        let is_mutation = document
+            .operations
+            .iter()
+            .any(|(_, operation)| operation.node.ty == OperationType::Mutation);
+        if is_mutation {
+            *self.is_mutation.lock().unwrap() = true;
+        }
```

Comment:
I would be more comfortable with us unconditionally setting the value here, to avoid situations where this  logic runs multiple times and we fail to set it to `false` from `true` for some reason.

```suggestion
        *self.is_mutation.lock().unwrap() = is_mutation;
```